t/mojo/collection.t

```diff
@@ -5,167 +5,186 @@ use Mojo::ByteStream qw(b);
+  is_deeply [@$collection], [1, 2, 3], 'right elements';
+  $collection = c(c(1, 2, 3), c(4, 5, 6), c(7, 8, 9));
+  is $collection->map('reverse')->map(join => "\n")->join("\n"), "3\n2\n1\n6\n5\n4\n9\n8\n7", 'right result';
+  is $collection->map(join => '-')->join("\n"), "1-2-3\n4-5-6\n7-8-9", 'right result';
+};
+
+subtest 'reverse' => sub {
+  my $collection = c(3, 2, 1);
+  is_deeply $collection->reverse->to_array, [1, 2, 3], 'right order';
+  $collection = c(3);
+  is_deeply $collection->reverse->to_array, [3], 'right order';
+  $collection = c();
+  is_deeply $collection->reverse->to_array, [], 'no elements';
+};
+
+subtest 'shuffle' => sub {
+  my $collection = c(0 .. 10000);
+  my $random = $collection->shuffle;
+  is $collection->size, $random->size, 'same number of elements';
+  isnt "@$collection", "@$random", 'different order';
+  is_deeply c()->shuffle->to_array, [], 'no elements';
+};
+
+subtest 'size' => sub {
+  my $collection = c();
+  is $collection->size, 0, 'right size';
+  $collection = c(undef);
+  is $collection->size, 1, 'right size';
+  $collection = c(23);
+  is $collection->size, 1, 'right size';
+  $collection = c([2, 3]);
+  is $collection->size, 1, 'right size';
+  $collection = c(5, 4, 3, 2, 1);
+  is $collection->size, 5, 'right size';
+};
+
+subtest 'reduce' => sub {
+  my $collection = c(2, 5, 4, 1);
+  is $collection->reduce(sub { $a + $b }), 12, 'right result';
+  is $collection->reduce(sub { $a + $b }, 5), 17, 'right result';
+  is c()->reduce(sub { $a + $b }), undef, 'no result';
+};
+
+subtest 'sort' => sub {
+  my $collection = c(2, 5, 4, 1);
+  is_deeply $collection->sort->to_array, [1, 2, 4, 5], 'right order';
+  is_deeply $collection->sort(sub { $b cmp $a })->to_array, [5, 4, 2, 1], 'right order';
+  is_deeply $collection->sort(sub { $_[1] cmp $_[0] })->to_array, [5, 4, 2, 1], 'right order';
+  $collection = c(qw(Test perl Mojo));
+  is_deeply $collection->sort(sub { uc(shift) cmp uc(shift) })->to_array, [qw(Mojo perl Test)], 'right order';
+  $collection = c();
+  is_deeply $collection->sort->to_array, [], 'no elements';
+  is_deeply $collection->sort(sub { $a cmp $b })->to_array, [], 'no elements';
+};
+
+subtest 'uniq' => sub {
+  my $collection = c(1, 2, 3, 2, 3, 4, 5, 4);
+  is_deeply $collection->uniq->to_array, [1, 2, 3, 4, 5], 'right result';
+  is_deeply $collection->uniq->reverse->uniq->to_array, [5, 4, 3, 2, 1], 'right result';
+  $collection = c([1, 2, 3], [3, 2, 1], [3, 1, 2]);
+  is_deeply $collection->uniq(sub { $_->[1] }), [[1, 2, 3], [3, 1, 2]], 'right result';
+  $collection = c(c(1, 2), c(1, 2), c(2, 1));
+  is_deeply $collection->uniq(join => ',')->flatten->to_array, [1, 2, 2, 1], 'right result';
+  $collection = c(undef, '', 3, 2, 1, 0);
+  is_deeply $collection->uniq->to_array, [undef, 3, 2, 1, 0], 'right result';
+  $collection = c(undef, '', 3, 2, 1, 0);
+  is_deeply $collection->uniq(sub {$_})->to_array, [undef, 3, 2, 1, 0], 'right result';
+};
+
+subtest 'TO_JSON' => sub {
```

Comment:
Subtest called `TO_JSON` but the method called is `encode_json`. IMO this should be changed to match the style of the other tests.

Reply:
`Mojo::Collection` contains a `TO_JSON` method that is actually tested here. All JSON encoders support that method.

Reply:
In my opinion it would be clearer exactly what is under test if that method was called directly, instead of the form that was chosen here. But I have no strong opinion that this needs changing, so I'll resolve this.